components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/queries/ConsentMgtCommonDBQueries.java

```diff
@@ -0,0 +1,416 @@
+
+        return "SELECT * FROM OB_CONSENT_AUTH_RESOURCE" + whereClause;
+    }
+
+    public String getStoreConsentStatusAuditRecordPreparedStatement() {
+
+        return "INSERT INTO OB_CONSENT_STATUS_AUDIT (STATUS_AUDIT_ID, CONSENT_ID, CURRENT_STATUS, ACTION_TIME, " +
+                "REASON, ACTION_BY, PREVIOUS_STATUS) VALUES (?, ?, ?, ?, ?, ?, ?)";
+    }
+
+    public String getGetConsentStatusAuditRecordsPreparedStatement() {
+
+        return "SELECT * FROM OB_CONSENT_STATUS_AUDIT WHERE CONSENT_ID = COALESCE(?, CONSENT_ID) " +
+                "AND CURRENT_STATUS = COALESCE(?, CURRENT_STATUS) AND ACTION_BY = COALESCE(?, ACTION_BY) " +
+                "AND STATUS_AUDIT_ID = COALESCE (?, STATUS_AUDIT_ID) AND ACTION_TIME >= COALESCE(?, ACTION_TIME) " +
+                "AND ACTION_TIME <= COALESCE(?, ACTION_TIME)";
+    }
+
+    /**
+     * SQL query for get consent status audit records by consentIds.
+     * @param whereClause conditions
+     * @param shouldLimit   whether to consider the Limit parameter
+     * @param shouldOffset  whether to consider the Offset parameter
+     * @return  SQL query for get consent status audit records by consentIds
+     */
+    public String getConsentStatusAuditRecordsByConsentIdsPreparedStatement(String whereClause, boolean shouldLimit,
+                                                                               boolean shouldOffset) {
+
+        StringBuilder query =
+                new StringBuilder("SELECT * FROM OB_CONSENT_STATUS_AUDIT " + whereClause);
+
+        if (shouldLimit && shouldOffset) {
+            query.append(" LIMIT ? OFFSET ? ");
+        } else if (shouldLimit) {
+            query.append(" LIMIT ? ");
+        }
+        return query.toString();
+    }
+
+    /**
+     * Util method to get the limit offset order for differentiate oracle and mssql pagination.
+     * @return is limit is before in prepared statement than offset
+     */
+    public boolean isLimitBeforeThanOffset() {
+
+        return true;
+    }
+
+    public String getInsertConsentHistoryPreparedStatement() {
+
+        return "INSERT INTO OB_CONSENT_HISTORY (TABLE_ID, RECORD_ID, HISTORY_ID, CHANGED_VALUES, " +
+                "REASON, EFFECTIVE_TIMESTAMP) VALUES (?, ?, ?, ?, ?, ?)";
+    }
+
+    public String getGetConsentHistoryPreparedStatement(String whereClause) {
+
+        return "SELECT * FROM OB_CONSENT_HISTORY " + whereClause + "ORDER BY EFFECTIVE_TIMESTAMP DESC";
+    }
+
+    public String getSearchExpiringConsentPreparedStatement(String statusesEligibleForExpirationCondition) {
+
+        return "SELECT OBC.CONSENT_ID " +
+                " FROM   OB_CONSENT_ATTRIBUTE CA " +
+                " JOIN   OB_CONSENT OBC " +
+                " ON     CA.CONSENT_ID = OBC.CONSENT_ID " +
+                " WHERE  CA.ATT_KEY = ? AND OBC.CURRENT_STATUS IN " + statusesEligibleForExpirationCondition;
+    }
+
+//    public String getGetConsentMappingResourcesForStatusPreparedStatement() {
```

Comment:
shall we remove these commented statements if these are not in use.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/2/commits/544143fd1d3ba2b3d581a41775d7e7bda95918ab